### DISCUSSION TITLE
Option to disable printing of generated text

### DISCUSSION THREAD

**Discussion by vineel96** [ggml-org/llama.cpp#10079] [Q&A] [answered]
Hi @ggerganov @danbev,
How to disable printing of generated output text to terminal? I could not find any command-line option?

**Comment by danbev** [comment]
If you are using `llama-cli` then it should be possible to use the `--log-disable` flag to avoid logging. For example:
```console
./llama-cli -m models/llama-2-7b.Q4_K_M.gguf --no-warmup --prompt '"Some prompt"' -n 10 --log-disable
```

**Reply by vineel96** [reply]
@danbev,
But --log-disable disables **all** output logs including perf timing data. I **want only the perf timing data** (i.e eval time, sample time etc) and I dont want rest all output(including generated text)

**Reply by danbev** [reply, marked as answer]
The logging of performance timing data is done using `info` log level which is also used in other places, and I'm not aware of any ability to have more fine grained logging. So just logging the perf data and nothing else is not possible at the moment as far as I know. But you can access this information as it is exposed in [llama.h](https://github.com/ggerganov/llama.cpp/blob/8f275a7c4593aa34147595a90282cf950a853690/include/llama.h#L1240) and adapt it to your use case.
